app/controllers/questionnaires_controller.rb

```diff
@@ -13,7 +13,7 @@ def action_allowed?
       (['Super-Administrator',
         'Administrator'].include? current_role_name) ||
           ((['Instructor'].include? current_role_name) && current_user_id?(@questionnaire.try(:instructor_id))) ||
-          ((['Teaching Assistant'].include? current_role_name) && assign_instructor_id == @questionnaire.try(:instructor_id))
+          ((['Teaching Assistant'].include? current_role_name) && session[:user].instructor_id == @questionnaire.try(:instructor_id))
```

Comment:
Does this work correctly for both Instructor and TA?

Reply:
Yes, it does.  If you look at the implementation, it has equivalent logic to what `assign_instructor_id` has.